File: src/main/java/com/appsolute/soomapi/domain/account/data/request/TeacherSignupRequest.java
package com.appsolute.soomapi.domain.account.data.request;

import com.appsolute.soomapi.domain.account.data.type.TeacherType;
import com.appsolute.soomapi.global.school.data.type.GenderType;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;

@Getter
@RequiredArgsConstructor
public class TeacherSignupRequest {
    private final String emailToken;
    private final String firstName;
    private final String lastName;
    private final GenderType gender;
    private final Integer grade;
    @DateTimeFormat(pattern = "yyMMdd")
    private final LocalDate birth;
    private final String rawPassword;
    private final TeacherType teacherType;
    private final String major;
    private final String code;
}

File: src/main/java/com/appsolute/soomapi/domain/account/data/deprecated/TeacherSignupRequest.kt
package com.appsolute.soomapi.domain.account.data.deprecated

import com.appsolute.soomapi.domain.account.data.type.Gender
import com.appsolute.soomapi.domain.account.data.entity.Teacher
import com.appsolute.soomapi.domain.account.data.type.TeacherType

@kotlin.Deprecated("domain.account.data.request.TeacherSignupRequest 를 사용해주세요! [지인호]", level = DeprecationLevel.ERROR)
data class TeacherSignupRequest(
    val email: String,
    val firstName: String,
    val lastname: String,
    val gender: Gender,
    val birth: String,
    val password: String,
    val teacherType: TeacherType,
    val major: String

) {
    fun toTeacher(id: String, password: String): Teacher{
        return Teacher(id, this.email, this.firstName, this.lastname, this.gender, this.birth, password, this.teacherType, this.major)

    }
}
